src/gui/shareusergroupwidget.cpp

```diff
@@ -926,13 +932,46 @@ void ShareUserLine::customizeStyle()
     _deleteShareButton->setIcon(deleteicon);
 
     _ui->noteConfirmButton->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
-    _ui->confirmExpirationDate->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
     _ui->progressIndicator->setColor(QGuiApplication::palette().color(QPalette::WindowText));
 
     // make sure to force BackgroundRole to QPalette::WindowText for a lable, because it's parent always has a different role set that applies to children unless customized
     _ui->errorLabel->setBackgroundRole(QPalette::WindowText);
 }
 
+QPixmap ShareUserLine::pixmapForShareeType(int type, const QColor &backgroundColor) const
+{
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+    switch (type) {
+    case Sharee::Room:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/talk-app.svg").arg(pixmapColor)));
+    case Sharee::Email:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/email.svg").arg(pixmapColor)));
+    default:
+        return {};
+    }
+}
+
+QColor ShareUserLine::backgroundColorForShareeType(int type) const
+{
+    switch (type) {
+    case Sharee::Room:
+        return Theme::instance()->wizardHeaderBackgroundColor();
+    case Sharee::Email:
+        return Theme::instance()->wizardHeaderTitleColor();
+    default:
+        break;
```

Comment:
break is not needed here I think. I would also suggest splitting the part above and the part below (if possible) into separate functions that explain what they do, because at least for me it is not really clear why we have for room and email a solid color and use the hash value for the rest :)

Reply:
the `break` is needed otherwise the compiler will refuse to compile the code